modules/db/src/main/java/org/mule/module/db/internal/result/statement/EagerStatementResultHandler.java

```diff
@@ -29,16 +31,28 @@ public EagerStatementResultHandler(ResultSetHandler resultSetHandler)
     }
 
     @Override
-    public Object processStatement(DbConnection connection, Statement statement, QueryTemplate queryTemplate, AutoGeneratedKeyStrategy autoGeneratedKeyStrategy) throws SQLException
+    public Map<String, Object> processStatement(DbConnection connection, Statement statement, QueryTemplate queryTemplate, AutoGeneratedKeyStrategy autoGeneratedKeyStrategy) throws SQLException
     {
+        Map<String, Object> resultsMap = null;
         try
         {
-            return super.processStatement(connection, statement, queryTemplate, autoGeneratedKeyStrategy);
+            resultsMap = super.processStatement(connection, statement, queryTemplate, autoGeneratedKeyStrategy);
+            return resultsMap;
         }
         finally
         {
             try
             {
+                if (resultsMap != null)
+                {
+                    for (Object resultValue : resultsMap.values())
+                    {
+                        if (resultValue instanceof ResultSet)
```

Comment:
I removed taht. Didn't realize that being eager there are no ResultSet's, only Map's